src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/DocumentsTest.java

```diff
@@ -18,14 +19,105 @@
 import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.Documents.getScannedDocuments;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.Documents.removeAlreadyAttachedDocuments;
 
 class DocumentsTest {
     private static final String SCANNED_DOCUMENTS = "scannedDocuments";
     private static final String DOCUMENT_NUMBER = "id";
 
+    @Test
+    void should_remove_all_documents_from_exception_record_because_all_of_them_are_already_in_target_case() {
+        // given
+        String exceptionRecordReference = "REF";
+        List<Map<String, Object>> exceptionRecordDocuments = singletonList(
+            singletonMap("value", singletonMap("controlNumber", "e-123"))
+        );
+        List<Map<String, Object>> targetCaseDocuments = singletonList(
+            singletonMap("value", ImmutableMap.of(
+                "controlNumber", "e-123",
+                "exceptionRecordReference", exceptionRecordReference
+            ))
+        );
+
+        // when
+        List<Map<String, Object>> actualDocumentsToAttach = removeAlreadyAttachedDocuments(
+            exceptionRecordDocuments,
+            targetCaseDocuments,
+            exceptionRecordReference
+        );
+
+        // then
+        assertThat(actualDocumentsToAttach).asList().isEmpty();
+    }
+
+    @Test
+    void should_remove_overlapping_documents_from_exception_record_because_they_are_already_in_target_case() {
+        // given
+        String exceptionRecordReference = "REF";
+        List<Map<String, Object>> exceptionRecordDocuments = ImmutableList.of(
+            singletonMap("value", singletonMap("controlNumber", "e-123")),
+            singletonMap("value", singletonMap("controlNumber", "e-321"))
+        );
+        List<Map<String, Object>> targetCaseDocuments = singletonList(
+            singletonMap("value", ImmutableMap.of(
+                "controlNumber", "e-123",
+                "exceptionRecordReference", exceptionRecordReference
+            ))
+        );
+
+        // when
+        List<Map<String, Object>> actualDocumentsToAttach = removeAlreadyAttachedDocuments(
+            exceptionRecordDocuments,
+            targetCaseDocuments,
+            exceptionRecordReference
+        );
+
+        // then
+        assertThat(actualDocumentsToAttach)
+            .asList()
+            .hasSize(1)
+            .first()
+            .isInstanceOfSatisfying(actualDocumentsToAttach.get(0).getClass(), document ->
+                assertThat(document)
+                    .hasFieldOrProperty("value")
+                    .extracting("value")
+                    .isInstanceOfSatisfying(document.getClass(), actualDocument ->
+                        assertThat(actualDocument)
+                            .hasFieldOrPropertyWithValue("controlNumber", "e-321")
+                    )
+            );
+    }
+
+    @Test
+    void should_leave_all_documents_in_exception_record_because_there_is_nothing_matching_in_target_case() {
```

Comment:
suggestion:  to be consistent because -> when

Reply:
can do. i think yesterday when i was typing it popped first as a sentence glue